src/Bundle/test/src/Tests/Controller/ScienceBookUiTest.php

```diff
@@ -56,6 +57,60 @@ public function it_allows_indexing_books(): void
         );
     }
 
+    /** @test */
+    public function it_allows_creating_a_book(): void
+    {
+        $newBookTitle = 'The Book of Why';
+        $newBookAuthorFirstName = 'Judea';
+        $newBookAuthorLastName = 'Pearl';
+
+        $this->loadFixturesFromFile('science_books.yml');
+
+        $this->client->request('GET', '/science-books/new');
+        $this->client->submitForm('Create', [
+            'sylius_resource[title]' => $newBookTitle,
+            'sylius_resource[author][firstName]' => $newBookAuthorFirstName,
+            'sylius_resource[author][lastName]' => $newBookAuthorLastName,
+        ]);
+
+        $this->assertResponseRedirects(null, expectedCode: Response::HTTP_FOUND);
+
+        /** @var ScienceBook $book */
+        $book = static::getContainer()->get('app.repository.science_book')->findOneBy(['title' => $newBookTitle]);
+
+        $this->assertNotNull($book);
+        $this->assertSame($newBookTitle, $book->getTitle());
+        $this->assertSame($newBookAuthorFirstName, $book->getAuthorFirstName());
+        $this->assertSame($newBookAuthorLastName, $book->getAuthorLastName());
```

Comment:
From my experience, it is better to stay on the same level of tests for the execution of tests and their validation.